src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,23 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  books: [],
+};
+
+const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, { payload }) => {
+      state.books.push(payload);
+    },
+    removeBook: (state, action) => {
+      const bookId = action.payload;
+      state.books.filter((book) => book.id !== bookId);
```

Comment:
- [ ] [OPTIONAL] Since `filter` is one of the array methods that does not mutate the original array, this line of code will not have any effect on the state. I think you have to assign it back to the state so it can work properly. 😎️

**Example**
```
state.books = state.books.filter((book) => book.id !== bookId);
```

Reply:
Thank you @amejid